---
manual:Javascript
version:0
lang:zh
rawUrl:https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RangeError
---





**RangeError**对象标明一个错误，当一个值不在其所允许的范围或者集合中。


## 语法<a name="语法"></a>

```
new RangeError([message[, fileName[, lineNumber]]])
```

### 参数<a name="参数"></a>
<dl><dt id=''>`message`</dt><dd>可选，可读的错误描述</dd><dt id=''>`fileName`<i></i></dt><dd>可选，包含造成异常代码的文件名</dd><dt id=''>`lineNumber`<i></i></dt><dd>可选，造成异常的代码所在的行数</dd></dl>
## 描述<a name="描述"></a>


试图传递一个number参数给一个范围内不包含该number的函数时则会引发**RangeError**。当传递一个不合法的length值作为[`Array`]6708 "entries() 方法返回一个新的Array Iterator对象，该对象包含数组中每个索引的键/值对。")构造器的参数创建数组，或者传递错误值到数值计算方法（[`Number.toExponential()`]41265 "toExponential() 方法以指数表示法返回该数值字符串表示形式。")，[`Number.toFixed()`]41266 "toFixed() 方法使用定点表示法来格式化一个数。")，[`Number.toPrecision()`]41267 "toPrecision() 方法以指定的精度返回该数值对象的字符串表示。")），会出现**RangeError**。.


## 属性<a name="属性"></a>
<dl><dt id=''>[`RangeError.prototype`]42329 "此页面仍未被本地化, 期待您的翻译!")</dt><dd>`允许在**RangeError**对象上附加属性。`</dd></dl>
## 方法<a name="方法"></a>


`RangeError`全局对象没有自带方法，但它通过可以原型链继承一些方法。


## `RangeError`实例<a name="RangeError实例"></a>

### 属性<a name="属性_2"></a>
<dl><dt id=''>`RangeError.prototype.constructor`</dt><dd>Specifies the function that created an instance&#39;s prototype.</dd><dt id=''>[`RangeError.prototype.message`]41994 "The message property is a human-readable description of the error.")</dt><dd>Error message. Although ECMA-262 specifies that[`RangeError`]40842 "The RangeError object indicates an error when a value is not in the set or range of allowed values.")should provide its own`message`property, in[SpiderMonkey]42011 ""), it inherits[`Error.prototype.message`]41994 "The message property is a human-readable description of the error.").</dd><dt id=''>[`RangeError.prototype.name`]41995 "The name property represents a name for the type of error. The initial value is "Error".")</dt><dd>Error name. Inherited from[`Error`]27232 "The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base object for user-defined exceptions. See below for standard built-in error types.").</dd><dt id=''>[`RangeError.prototype.fileName`]41998 "The fileName property contains the path to the file that raised this error.")</dt><dd>Path to file that raised this error. Inherited from[`Error`]27232 "The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base object for user-defined exceptions. See below for standard built-in error types.").</dd><dt id=''>[`RangeError.prototype.lineNumber`]41999 "The lineNumber property contains the line number in the file that raised this error.")</dt><dd>Line number in file that raised this error. Inherited from[`Error`]27232 "The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base object for user-defined exceptions. See below for standard built-in error types.").</dd><dt id=''>[`RangeError.prototype.columnNumber`]42000 "The columnNumber property contains the column number in the line of the file that raised this error.")</dt><dd>Column number in line that raised this error. Inherited from[`Error`]27232 "The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base object for user-defined exceptions. See below for standard built-in error types.").</dd><dt id=''>[`RangeError.prototype.stack`]42001 "The non-standard stack property of Error objects offer a trace of which functions were called, in what order, from which line and file, and with what arguments. The stack string proceeds from the most recent calls to earlier ones, leading back to the original global scope call.")</dt><dd>Stack trace. Inherited from[`Error`]27232 "The Error constructor creates an error object. Instances of Error objects are thrown when runtime errors occur. The Error object can also be used as a base object for user-defined exceptions. See below for standard built-in error types.").</dd></dl>

### 方法<a name="方法_2"></a>


Although the[`RangeError`]40842 "The RangeError object indicates an error when a value is not in the set or range of allowed values.")prototype object does not contain any methods of its own,[`RangeError`]40842 "The RangeError object indicates an error when a value is not in the set or range of allowed values.")instances do inherit some methods through the prototype chain.



## 例子<a name="例子"></a>

### 使用`RangeError`<a name="使用RangeError"></a>

```
var check = function(num) {
  if (num < MIN || num > MAX) {
    throw new RangeError('Parameter must be between ' + MIN + ' and ' + MAX);
  }
};

try {
  check(500);
}
catch (e) {
  if (e instanceof RangeError) {
    // 处理越界错误
  }
}
```

## 规范<a name="规范"></a>

Specification | Status | Comment 
[ECMAScript 3rd Edition (ECMA-262)]42013 "ECMAScript 3rd Edition (ECMA-262)") | Standard | Initial definition. 
[ECMAScript 5.1 (ECMA-262)<br></br><small>RangeError</small>]42330 "") | Standard |  
[ECMAScript 2015 (6th Edition, ECMA-262)<br></br><small>RangeError</small>]42331 "") | Standard |  
[ECMAScript Latest Draft (ECMA-262)<br></br><small>RangeError</small>]42332 "") | Draft |  


## 浏览器兼容性<a name="浏览器兼容性"></a>


**[We&#39;re converting our compatibility data into a machine-readable JSON format]3344 "")**. This compatibility table still uses the old format, because we haven&#39;t yet converted the data it contains.**[Find out how you can help!]3392 "")**


* 
* 

Feature | Chrome | Firefox (Gecko) | Internet Explorer | Opera | Safari 
Basic support | (Yes) | (Yes) | (Yes) | (Yes) | (Yes) 





## 相关连接<a name="相关连接"></a>

* [`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")
* [`RangeError.prototype`]42329 "此页面仍未被本地化, 期待您的翻译!")
* [`Array`]6708 "entries() 方法返回一个新的Array Iterator对象，该对象包含数组中每个索引的键/值对。")
* [`Number.toExponential()`]41265 "toExponential() 方法以指数表示法返回该数值字符串表示形式。")
* [`Number.toFixed()`]41266 "toFixed() 方法使用定点表示法来格式化一个数。")
* [`Number.toPrecision()`]41267 "toPrecision() 方法以指定的精度返回该数值对象的字符串表示。")



